File: packages/core-python/payafrica/payafrica.py
from __future__ import annotations

from .contracts import PaymentProvider, WebhookHeaders
from .models import PaymentEvent, PaymentRequest, PaymentSession, PaymentStatusResult, RefundResult


class PayAfrica:
    def __init__(self, provider: PaymentProvider) -> None:
        self._provider = provider

    async def initiate_payment(self, params: PaymentRequest) -> PaymentSession:
        return await self._provider.initiate_payment(params)

    async def check_status(self, session_id: str) -> PaymentStatusResult:
        return await self._provider.check_status(session_id)

    async def handle_webhook(self, raw_body: str | bytes, headers: WebhookHeaders) -> PaymentEvent:
        return await self._provider.handle_webhook(raw_body, headers)

    async def refund(self, session_id: str, amount: int | float | None = None) -> RefundResult:
        return await self._provider.refund(session_id, amount)

File: packages/core-node/src/PayAfrica.ts
import type {
  PaymentEvent,
  PaymentProvider,
  PaymentRequest,
  PaymentSession,
  PaymentStatus,
  PaymentStatusResult,
  RefundResult,
} from "./types.js";

export class PayAfrica {
  public constructor(private readonly provider: PaymentProvider) {}

  public initiatePayment(params: PaymentRequest): Promise<PaymentSession> {
    return this.provider.initiatePayment(params);
  }

  public checkStatus(sessionId: string): Promise<PaymentStatusResult> {
    return this.provider.checkStatus(sessionId);
  }

  public handleWebhook(
    rawBody: string | Buffer,
    headers: Record<string, string | string[] | undefined>
  ): Promise<PaymentEvent> {
    return this.provider.handleWebhook(rawBody, headers);
  }

  public refund(sessionId: string, amount?: number): Promise<RefundResult> {
    return this.provider.refund(sessionId, amount);
  }
}
